File: apps/gateway/src/orders/schemas.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class CreateOrder(BaseModel):
    product_id: str
    user_email: str
    quantity: int
    price: float
    currency: str
    user_id: str


class DisplayOrder(BaseModel):
    id: UUID
    product_id: str
    user_email: str
    quantity: int
    price: float
    currency: str
    user_id: str
    status: str
    created_at: datetime
    updated_at: datetime

File: apps/orders/src/core/schemas.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Serialize, Deserialize)]
pub struct Order {
    pub id: Uuid,
    pub product_id: String,
    pub user_email: String,
    pub quantity: u32,
    pub price: f64,
    pub currency: String,
    pub user_id: String,
}
